core/state_transition.go

```diff
@@ -151,10 +152,19 @@ type Message struct {
 }
 
 // TransactionToMessage converts a transaction into a Message.
-func TransactionToMessage(tx *types.Transaction, s types.Signer, baseFee, headerGasTip *big.Int) (*Message, error) {
+func TransactionToMessage(tx *types.Transaction, s types.Signer, baseFee, headerGasTip *big.Int, statedb *state.StateDB) (*Message, error) {
+	from, err := types.Sender(s, tx)
+	if err != nil {
+		return nil, err
+	}
+
 	gasTipCap := new(big.Int).Set(tx.GasTipCap())
 	if headerGasTip != nil {
-		gasTipCap = new(big.Int).Set(headerGasTip)
+		// If Anzeon is enabled, and the sender is authorized, use the tx's tx.GasTipCap()
+		// Otherwise, use the header's gas tip
+		if statedb != nil && !statedb.IsAuthorized(from) {
+			gasTipCap = new(big.Int).Set(headerGasTip)
+		}
```

Comment:
```suggestion
		if statedb == nil {
			gasTipCap = new(big.Int).Set(headerGasTip)
		} else if !statedb.IsAuthorized(from) {
			gasTipCap = new(big.Int).Set(headerGasTip)
		}
```

If normal account and statedb is nil, gasTipCap is different with headerGasTip
Please refer to the test code in state_processor_test.go

Reply:
I talked with @hominlee-wemade . This is not issue.